### PR TITLE
hidden/visible-selector: add note about performance

### PR DESCRIPTION
**Pull request by arthurvr**
Would fix gh-679

### CODE REVIEW

**Comment by scottgonzalez** [comment]
In my opinion, this is too strongly worded. There are many sane uses of these selectors that don't warrant a strong warning.

**Comment by dmethvin** [comment]
It's actually pretty weasel worded as it is. What alternate wording would you suggest?

File: notes.xsl [hunk 1]
```diff
@@ -58,6 +58,9 @@
 		<xsl:when test="@id = 'html-code-execution'">
 			By design, any jQuery constructor or method that accepts an HTML string — <a href="/jQuery/">jQuery()</a>, <a href="/append/">.append()</a>, <a href="/after/">.after()</a>, etc. — can potentially execute code. This can occur by injection of script tags or use of HTML attributes that execute code (for example, <code>&lt;img onload=""&gt;</code>). Do not use these methods to insert strings obtained from untrusted sources such as URL query parameters, cookies, or form inputs. Doing so can introduce cross-site-scripting (XSS) vulnerabilities. Remove or escape any user input before adding content to the document.
 		</xsl:when>
+		<xsl:when test="@id = 'hidden-forces-layout'">
+			Using this selector heavily has some performance regressions, as it may force the browser to re-render the page before it can determine visibility. Tracking the visibility of elements via other methods, for example using a class, can provide better performance.
```
Comment (kswedberg) [inline notes.xsl]: Not sure "regressions" is the right word here. What is it regressing from? Maybe, especially considering @scottgonzalez's concerns, we could change it to "implications"?

Comment (arthurvr) [inline notes.xsl]: @kswedberg Yup. Will update it.

**Comment by timmywil** [comment]
How about "can have performance implications" instead of "has some performance implications"?

**Comment by scottgonzalez** [comment]
> How about "can have performance implications"

+1

**Comment by arthurvr** [comment]
:+1:. I updated the PR. Looking good now?

**Comment by scottgonzalez** [comment]
"Using this selector heavily" sounds odd. Do you mean things like using this inside a loop?

**Comment by dmethvin** [comment]
It could be inside a loop where the DOM is being changed and thus forcing recalc each time, or could be just used once or twice in a very large DOM.

**Comment by arthurvr** [comment]
Any suggestions are welcome, though I think it's quite okay.

**Comment by dmethvin** [comment]
I think it's good now. :+1:

**Pull request closed without merging**